File: Python/0061. Page Number.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            n,k = map(int,input().split())
            if n<7:
                for i in range(1,n):
                    print(i," ",sep='',end='')
                print(n,end='')
            
            else:
                print(1,end=' ')
                if k<=4:
                    for i in range(2,k+2+1):
                        print(i," ",sep='',end='')
                else:
                    print("... ",end='')
                    for i in range(k-2,min(k+2,n-1)+1):
                        print(i," ",sep='',end='')

                if k<n-3:
                    print("... ",sep='',end='')
                print(n,sep='',end='')
            
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0061. Page Number.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    int n,k;
    while(cin>>n>>k){
        if(n<7){
            for(int i=1;i<n;++i){
                cout<<i<<" ";
            }
            cout<<n;
        }
        else{
            cout<<1<<" ";
            if(k<=4){
                for(int i=2;i<=k+2;++i){
                    cout<<i<<" ";
                }
            }
            else{
                cout<<"... ";
                for(int i=k-2;i<=min(k+2,n-1);++i){
                    cout<<i<<" ";
                }
            }
            if(k<n-3){
                cout<<"... ";
            }
            cout<<n<<endl;
        }
    }
    return 0;
}
